readme.md

```diff
@@ -44,7 +44,7 @@ Yields:
 
 ## API
 
-### visit([node](https://github.com/wooorm/unist#unist-nodes)\[, type\], callback\[, reverse\])
+### visit([node](https://github.com/wooorm/unist#unist-nodes)\[, type], callback\[, reverse])
```

Comment:
> (And BTW, remark-man doesn't seem to follow this convention.)

The fact that remark-man removes the brackets is because of a bug, I think, in Roff pages those should be escaped but I forgot to fix that :)

Anyway, **I** was delusional in my above statements, and was missing some stuff already in place:

```md
### visit([foo])

### visit(\[bar\])
```

...is processed to:

```md
### visit([foo])

### visit(\[bar])
```

So:

* Things read as reference links (even without definition), are kept that way (I think I was missing this);
* Escaped reference links (`\[foo]`) are kept this way, seen as `"[", "foo]"` in the tree (two text nodes), and compiled escaped;
* The superfluous escape for the closing bracket is removed `\[foo\]` > `\[foo]`, as there’s no need to escape both (also fine).

You are right, my proposed solution, removing escapes when not needed would and thus resulting in references, interferes with possible future definitions. I was wrong, sorry! :anguished: